src/modules/hip/kernel/brightness.hpp

```diff
@@ -0,0 +1,288 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+
+__device__ void brightness_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, float4 *alpha_f4, float4 *beta_f4)
+{
+    dst_f8->x = src_f8->x * *alpha_f4 + *beta_f4;
+    dst_f8->y = src_f8->y * *alpha_f4 + *beta_f4;
+}
+
+__device__ void brightness_hip_compute(float *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, float4 *alpha_f4, float4 *beta_f4)
+{
+    dst_f8->x = src_f8->x * *alpha_f4 + *beta_f4 * (float4)0.0039216;
+    dst_f8->y = src_f8->y * *alpha_f4 + *beta_f4 * (float4)0.0039216;
+}
+
+__device__ void brightness_hip_compute(signed char *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, float4 *alpha_f4, float4 *beta_f4)
+{
+    dst_f8->x = rpp_hip_pixel_check((src_f8->x + 128) * *alpha_f4 + *beta_f4) - 128;
```

Comment:
src_f8->x is a float4 so if your intention is to add each of the x/y/z/w with 128 then you need to use (float4)128 here.

similarly, you would need to use (float4)128 to subtract in the end of this line.

Reply:
Thats true, but I believe it was doing an auto typecast to float4. In any case, I have added a manual typecast now to be on the safer side.